madminer/utils/morphing.py

```diff
@@ -113,80 +119,186 @@ def set_components(self, components):
         self.components = components
         self.n_components = len(self.components)
 
-    def find_components(self, max_overall_power=4):
+    def find_components(self, max_overall_power=0, BSM_max_power = float('inf'), Nd = 0, Np = 0, Ns = 0):
         """
         Finds the components, i.e. the individual terms contributing to the squared matrix element.
-
+        
         Parameters
         ----------
-        max_overall_power : int, optional
-            The maximal sum of powers of all parameters contributing to the squared matrix element.
-            Typically, if parameters can affect the couplings at n vertices, this number is 2n. Default value: 4.
+        BSM_max_power : int, optional, default = float('inf')
+            To set the maximal power for BSM Couplings
+        Nd : int, optional, default = 0
+            Number of decay couplings
+        Np : int, optional, default = 0
+            Number of production couplings
+        Ns : int, optional, default = 0
+            Number of spin couplings
```

Comment:
```suggestion
            Number of same couplings
```
We might need more explanations either here or elsewhere to clarify what this means.

Reply:
Yes I'll add that it stands for both production and decay